lib/facility.rb

```diff
@@ -1,14 +1,73 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name,
+              :address,
+              :phone,
+              :services,
+              :registered_vehicles,
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility)
+    @name = facility[:name]
+    @address = facility[:address]
+    @phone = facility[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    
+    if @services.include?('Vehicle Registration') == true
+      @registered_vehicles << vehicle
+      vehicle.set_registration_date
+      vehicle.set_plate_type
+
+      if vehicle.antique? == true
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle? == true
+        @collected_fees += 200
+      else
+        @collected_fees += 100
+      end
+    end
+
+    if @registered_vehicles.empty? == true
+      return nil
+    elsif @registered_vehicles.empty? == false
+      return @registered_vehicles
+    end
+    
+  end
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test') == true
+      if registrant.age >= 16 && registrant.permit == true
+        registrant.license_data[:written] = true
+      end
+    end
+    return registrant.license_data[:written]
+  end
+
+  def administer_road_test(registrant)
+    if @services.include?('Road Test') == true
```

Comment:
Consider chaining together the conditional checks here with &&
```
if @services.include?('Renew License') == true && registrant.license_data[:license] == true
        registrant.license_data[:renewed] = true
      end

```

Reply:
Again, you dont need to explictly check that these values == true, as `include?` will evaluate to `true` or `false` inherinetly